packages/core/src/utils.ts

```diff
@@ -975,24 +995,47 @@ export const toOpenZeppelinProxyType = (
 }
 
 export const getOpenZeppelinValidationOpts = (
-  proxyType: ProxyType
+  proxyType: ProxyType,
+  contractConfig: UserContractConfig
 ): Required<ValidationOptions> => {
-  return {
+  type UnsafeAllow = Required<ValidationOptions>['unsafeAllow']
+
+  const unsafeAllow: UnsafeAllow = [
+    'state-variable-assignment',
+    'constructor',
+    'state-variable-immutable',
+    'missing-public-upgradeto',
+  ]
+  if (contractConfig.unsafeAllow?.delegatecall) {
+    unsafeAllow.push('delegatecall')
+  }
+  if (contractConfig.unsafeAllow?.selfdestruct) {
+    unsafeAllow.push('selfdestruct')
+  }
+
+  const options = {
     kind: toOpenZeppelinProxyType(proxyType),
-    unsafeAllow: [],
+    unsafeAllow,
     unsafeAllowCustomTypes: false,
```

Comment:
We can remove this line since `unsafeAllowCustomTypes` defaults to false in `withValidationDefaults`